crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -20,18 +20,39 @@ module Api
   class Upgrade < Tableless
     def status
       {
-        crowbar: crowbar_upgrade_status,
-        checks: check
-      }
+        crowbar: crowbar_upgrade_status
+      }.merge!(check)
     end
 
     def check
       {
-        sanity_checks: sanity_checks,
-        maintenance_updates_installed: maintenance_updates_installed?,
-        clusters_healthy: clusters_healthy?,
-        compute_resources_available: compute_resources_available?,
-        ceph_healthy: ceph_healthy?
+        checks: {
+          sanity_checks: {
+            required: true,
+            passed: sanity_checks.empty?,
+            errors: sanity_check_errors
+          },
+          maintenance_updates_installed: {
+            required: true,
+            passed: maintenance_updates_status[:passed],
```

Comment:
we have had a short meeting with the ui team, and they don't require a separate warnings structure. also the example you chose here is maybe not the best, because the maintenance updates check already refreshes the metadata.